### PR TITLE
Add new border styles

### PR DESCRIPTION
**Pull request by LaurentShala**
# Details
This PR adds the ability to update the border style of the buttons to have the old style (border all around button) and the new style (border on the bottom only)

This PR creates a subclass of UIButton called `PinButton` which contains a state.

By default, the behavior and UI looks as before with the `full` border style being set by default.

# Screenshots
## `.bottom` border style (New Style)
group
## `.full` border style (Old Style)
group 2

### CODE REVIEW

**Comment by jeffaburt** [comment]
Since this is open-source, I challenge you to find a way to opt in to the new style, so you don't break existing installs that use "old style" or new installs that prefer "old style". Also, you should update the README.md with screenshots and examples of how to use both styles.

**Comment by jeffaburt** [comment]
Also, consider updating the example app with a toggle switch to transition between new and old styles.

**Comment by LaurentShala** [comment]
Challenge accepted 😉

**Comment by LaurentShala** [comment]
@jeffaburt Check it out homie 💪

**Comment by jeffaburt** [comment]
I don't see any README.md updates, homie.

File: PinEntryView/Classes/PinEntryView.swift [hunk 1]
```diff
@@ -350,24 +353,31 @@ fileprivate extension PinEntryView {
...
+            buttonState.borderStyle = state?.buttonBorderStyle ?? .full
+            
             if let newCharacter = textField.text?[i],
                 newCharacter != "" {
-                button.setTitle(newCharacter, for: .normal)
-                button.setTitleColor(.black, for: .normal)
-                button.layer.borderColor = state?.completedBorderColor.cgColor
+                
+                buttonState.title = newCharacter
+                buttonState.textColor = state?.completedBorderColor ?? .black
```
Comment (jeffaburt) [inline PinEntryView/Classes/PinEntryView.swift]: Noticed you changed this from just being .black. Intentional?

**Comment by jeffaburt** [comment]
FYI I pushed version 3.0.0 to CocoaPods with these changes 🎉 

image

**Comment by jrmsklar** [comment]
🚀🚀

**Pull request merged**